org.eclipse.xtext.ui.tests/src-gen/org/eclipse/xtext/ui/tests/editor/contentassist/services/Bug381381TestLanguageGrammarAccess.java

```diff
@@ -66,18 +66,18 @@
 		private final Keyword cREPLACEKeyword_1_2_3 = (Keyword)cGroup_1_2.eContents().get(3);
 		
 		//CopyFieldNameToVariableStmt:
-		//	'FIELD-NAME-TO-VARIABLE' (',' 'SCREEN' '=' '(' line=INT ',' column=INT ')' & ',' 'VAR' '=' name=ID & (',' 'TYPE' '='
-		//	'REPLACE')?);
+		//	'FIELD-NAME-TO-VARIABLE' ( ',' 'SCREEN' '=' '(' line=INT ',' column=INT ')' & ',' 'VAR' '=' name=ID & (',' 'TYPE' '='
+		//	'REPLACE')? );
```

Comment:
@ArneDeutsch  The space before the closing parentheses is unexpected.

Reply:
Should be fixed with https://github.com/eclipse/xtext-core/issues/1489

Reply:
problem still seems to be there

Reply:
I do not really understand why the issue is still there. I have added a dedicated test case (`JavaFormatterGrammarTest.unorderedGroups`). It is succeeding, formatting is correct. But on the otehr hand the generated comment is clearly wrong ... it is not only the whitespace, there is a missing parantesis, too.

Original Code: (notice the two openeing parentesis before the `','`.
```
CopyFieldNameToVariableStmt:
  'FIELD-NAME-TO-VARIABLE' ((',' 'SCREEN' '=' '(' line=INT ',' column=INT ')') &
                            (',' 'VAR' '=' name=ID) &
                            (',' 'TYPE' '=' 'REPLACE')?);
```

In my test this is formatted to:
```
CopyFieldNameToVariableStmt:
	'FIELD-NAME-TO-VARIABLE' ((',' 'SCREEN' '=' '(' line=INT ',' column=INT ')') &
	(',' 'VAR' '=' name=ID) &
	(',' 'TYPE' '=' 'REPLACE')?);
```

But the comment in the end looks like this:
```
//CopyFieldNameToVariableStmt:
//	'FIELD-NAME-TO-VARIABLE' ( ',' 'SCREEN' '=' '(' line=INT ',' column=INT ')' & ',' 'VAR' '=' name=ID & (',' 'TYPE' '='
//	'REPLACE')? );
```

For me it looks like a bug in the serializer, because of the missing parantesis ... am a bit clueless here ...

Reply:
I created a ticket for this one: https://github.com/eclipse/xtext-core/issues/1641